processing/src/main/java/io/druid/query/aggregation/post/ArithmeticPostAggregator.java

```diff
@@ -47,25 +48,40 @@ public int compare(Object o, Object o1)
   private final String fnName;
   private final List<PostAggregator> fields;
   private final Ops op;
+  private final Comparator comparator;
+  private final String ordering;
+
+  public ArithmeticPostAggregator(
+      String name,
+      String fnName,
+      List<PostAggregator> fields
+  )
+  {
+    this(name, fnName, fields, null);
+  }
 
   @JsonCreator
   public ArithmeticPostAggregator(
       @JsonProperty("name") String name,
       @JsonProperty("fn") String fnName,
-      @JsonProperty("fields") List<PostAggregator> fields
+      @JsonProperty("fields") List<PostAggregator> fields,
+      @JsonProperty("ordering") String ordering
   )
   {
+    Preconditions.checkArgument(fnName != null, "fn cannot not be null");
+    Preconditions.checkArgument(fields != null && fields.size() > 1, "Illegal number of fields[%s], must be > 1");
+
     this.name = name;
     this.fnName = fnName;
     this.fields = fields;
-    if (fields.size() <= 1) {
-      throw new IAE("Illegal number of fields[%s], must be > 1", fields.size());
-    }
 
     this.op = Ops.lookup(fnName);
     if (op == null) {
       throw new IAE("Unknown operation[%s], known operations[%s]", fnName, Ops.getFns());
     }
+
+    this.ordering = ordering;
+    this.comparator = ordering == null ? DEFAULT_COMPARATOR : Ordering.valueOf(ordering);
```

Comment:
I think whatever default we choose people will complain and the feature that is really needed is some notion of "having" for post aggs. With that said, I think we should show +INF in the results just to let people know the query they are making has wacky results. However, this changes default behavior and may cause problems for existing deployments.